packages/eslint-config-4catalyzer-typescript/rules.js

```diff
@@ -14,6 +14,15 @@ module.exports = {
   'import/default': 'off',
   'import/namespace': 'off',
 
+  'no-shadow': 'off',
+  '@typescript-eslint/no-shadow': [
+    'error',
+    {
+      ignoreTypeValueShadow: true,
+      ignoreFunctionTypeParameterNameValueShadow: true,
```

Comment:
why would we ignore these? they both seem legit. I had to fix my code to comply but it made sense

Reply:
the first one we do intentionally in places and i didn't want to push another breaking change right now.
The function one seems actively wrong no? You want to be able to name function type parameters whatever is accurate, shadowing doesn't cause bugs in those cases. 

like why does it matter that `test` is used twice here? a consumer of the fn type isn't going to see anything else and giving it a worse name makes the type less clear to users of the type intellisense
```ts
const test = 1;
type Func = (test: string) => typeof test;
```

Reply:
> why does it matter that `test` is used twice here?

for the same reason shadowing in general matters, I guess. I am fine if you are not a fan of this though!

Reply:
I think it's more convenient to allow it here! happy to revisit if it causes bugs